### ISSUE TITLE
Ability for end users to load data on to prod and treat the data as a source of truth for others to consume

### ISSUE THREAD

**Issue by anandsiva** [CommBank/maestro#437]
- [ ] End user should be able to upload the data via portal
- [ ]  data is validated against the underlying structure.
- [ ] Ability for the the end user to add or modify the structure of the table
- [ ] Only authorised users are allowed to make changes.
- [ ] Users should be able to upload data only to the tables were they were assigned access to load
- [ ] Audit log of all the changes to be captured.
- [ ] Users should be able to upload the data  only after validating the data against the schema of the table
- [ ] Should be able to support standard csv  format files.
- [ ]  Any schema validation error should be thrown in the UI with reference to the line number and column in the data which lead to the error in the first place limiting to a max of 10 errors.
- [ ] Once the data is submitted the workflow should go to the data owner for review and approval. Only approved data upload gets published.
- [ ] any data upload that is rejected should send out appropriate email alert to the requester. The approver while rejection should have to provide justification for the reject and that becomes part of the email.
- [ ]  when a requester request to upload the data an email is sent to approver with the link to take appropriate action.
- [ ]  The entire solution/portal should be over SSL.

**Comment by anandsiva** [comment]
Rationale:-
For user data to be treated as production data we need to have:-
- an automated process which the production support team can support.
- audit traceability of who and when some one had loaded the data.
- prevent unauthorised users making changes to the data
- data is validated for correctness ( Through code rather than manual verification ).
- appropriate security restrictions in place.
- the user should not be able to upload to other data assets which are loaded from source systems.

**Comment by stephanh** [comment]
@anandsiva can you raise this issue here https://github.com/CommBank/maestro-team. This is a larger issues that won't be directly provided by the Maestro library but is instead work that needs to be done by the team. We will probably stand up a separate service and repo for this.

**Comment by anandsiva** [comment]
Yes. Created a new ticket. https://github.com/CommBank/maestro-team/issues/11
